docs/design/fabric-v2+/fpc-cc-invocation.puml

```diff
@@ -141,35 +146,35 @@ group invoke
       Enclave1 -> Enclave1 : ccrsp_msg.enclave_id <- Enclave_Id
```

Comment:
Since the enclave creates the endorsement signature using `args, read/writeset, and result_enc` we need to ensure that all these values are available when chaincode-hosted endorsement is performed. In particular, our current flow misses the `args`. With traditional custom validation, we have full access of the transaction proposal, which contains the entire proposal. With our new validation, we only send the `ccresp_msg` to validation, hence, we are missing the `args`.

I suggest that we just add the `proposal` to `ccresp_msg`.

Reply:
Also, instead of signing just the `args`, why don't we include the entire proposal in the signature?

Reply:
good find: besides obviously needing the parameters to verify, we also should sign the proposal to capture also the creator which can influence the chaincode and hence should be audited ..